lib/discovery.js

```diff
@@ -380,43 +380,59 @@ export function isAccountEmpty(account: AccountDiscoveryState): boolean {
 export type AccountDiscoveryData = {
     sources: Array<CachingSourceData>;
     histories: Array<ChainHistory>;
-    transactions: TransactionMapData;
+    transactionIds: Array<string>;
     blocks: BlockRange;
 };
 
+export type AccountDiscoveryDataWithTransactions = {
+    data: AccountDiscoveryData;
+    getTransaction: (id: string) => Promise<TransactionInfoData>;
+}
+
 export function storeAccountDiscovery(
     account: AccountDiscoveryState,
     sources: Array<CachingSource>
-): AccountDiscoveryData {
+): AccountDiscoveryDataWithTransactions {
     let t0 = account[0].transactions;
     let t1 = account[1].transactions;
-    return {
+    let txs = t0.merge(t1);
```

Comment:
Maps are usually already merged at this point, it would be cool to re-use the merge.